pallets/subtensor/src/staking/stake_utils.rs

```diff
@@ -852,9 +860,10 @@ impl<T: Config> Pallet<T> {
         // there's no slippage in this move)
         let current_price =
             <T as pallet::Config>::SwapInterface::current_alpha_price(netuid.into());
-        let tao_equivalent = current_price
+        let tao_equivalent: TaoCurrency = current_price
             .saturating_mul(U96F32::saturating_from_num(actual_alpha_moved))
-            .saturating_to_num::<u64>();
+            .saturating_to_num::<u64>()
+            .into();
```

Comment:
Could be cool to impl `From` for these floating point types so we can skip the intermediary u64 step. not a blocker